src/sentry/static/sentry/app/views/settings/organizationAuth/index.jsx

```diff
@@ -90,18 +93,27 @@ class OrganizationAuth extends AsyncView {
 
   renderBody() {
     let {providerList, provider} = this.state;
+    let access = this.context.organization.access;
 
-    if (!provider) {
-      return (
+    if (access.includes('org:admin') && provider) {
+      // If SSO provider is configured, keep showing loading while we redirect
+      // to django configuration view
+      return this.renderLoading();
+    }
+
+    let activeProvider = providerList.find(
+      p => provider && p.key === provider.provider_name
+    );
+
+    return (
+      <React.Fragment>
```

Comment:
Why do you need a wrapping fragment? Is `OrganizationAuthList` not a normal component?

Reply:
Good point. got rid of it.